ships/0021-local-source-upload.md

```diff
@@ -0,0 +1,363 @@
+means a simple application meant to wait for a specific condition, and exit gracefully afterwards.
+Here we may keep checking if a given "lock" file exists, when removed the application exists. It also
+must exit with an error on timeout.
+
+Additionally, we can employ this application to make verifications on the streamed data, like for
+instance checksum, ownership (UID/GID) and more.
+
+The implementation written for research and development can be found at [otaviof/waiter][Waiter]
+repository, and it is started by running "`waiter start`". When running it will keep watching the
+file `/var/tmp/waiter.pid`, and when removed, it will exit gracefully. The following example shows it
+running in the container-manager. In the following example the entrypoint receives the argument
+"`start`", and the waiting begins.
+
+```sh
+$ docker run --name=waiter --interactive otaviof/waiter:latest start
+2021/06/01 08:25:24 Saving PID '1' at '/var/tmp/waiter.pid'
+2021/06/01 08:25:24 Attempts '600'...
+2021/06/01 08:25:25 Attempts '599'...
+2021/06/01 08:25:26 Attempts '598'...
+2021/06/01 08:25:27 Done!
+```
+
+When the data transfer is complete, the CLI will execute "`waiter done`", via the same "`kubectl
+exec`" mechanism.
+
+```sh
+$ docker exec waiter waiter done
+2021/06/01 08:25:56 Removing pid-file '/var/tmp/waiter.pid' (1)
+```
+
+This procedure works on the current version of [Tekton (v0.22.0)][TektonV0220] used in Shipwright
+Build-Controller.
+
+##### Waiter on Controller Repository
+
+The `waiter` application will live on the [Build Controller repository][ControllerCmd], side by side
+with other applications, and producing a stand alone container-image. The image must be based on
+UBI8, following the same standards than others.
+
+This image also has `tar` installed, which will be used for the data upload mechanism, as explained
+before.
+
+Another requirement is following the same configuration standard than other external images used by
+the Build Controller, like a environment variable (i.e.: `GIT_CONTAINER_TEMPLATE`) to overwrite the
+image before the controller starts.
+
+At last, it should provide meaningful information for Tekton's `TaskResult`, so we can differentiate
+this step run from others. For the `waiter` process results, it can take advantage of the
+[termination-log pattern][KubernetesTerminationLog] to inform Tekton more detailed execution results.
+
+### Use-Cases
+
+#### Developer's Inner-Loop
+
+As a software developer, I'm able to try out the local changes being worked on with a container-image
+built by the same Shipwright Build-Controller and using cluster's computing power. Another positive
+side effect is that developers won't need to install a container-manager, like for instance Docker,
+to be able to build new images.
+
+#### Continuous-Integration
+
+During CI pipelines Shipwright Build-Controller can be employed to build container-images based on
```

Comment:
this also feels like a user story to mention up top in a user story section per the latest template ... I don't remember you mentioning this one up top in the other sections, like you did with the "as a dev I want to try out locally first" one

Reply:
I'm using the section as "user stories", as you suggested. Indeed, makes sense.